Userland/Applications/Magnifier/MagnifierWidget.cpp

```diff
@@ -53,7 +53,15 @@ void MagnifierWidget::sync()
 
     auto size = frame_inner_rect().size();
     Gfx::IntSize grab_size { size.width() / m_scale_factor, size.height() / m_scale_factor };
-    m_grabbed_bitmap = GUI::ConnectionToWindowServer::the().get_screen_bitmap_around_cursor(grab_size).bitmap();
+    if (m_lock_location && !m_locked_location.has_value()) {
+        m_locked_location = GUI::ConnectionToWindowServer::the().get_global_cursor_position();
```

Comment:
Is there a reason this can't happen in `lock_location(true)` instead?

If it did, you wouldn't need `m_lock_location`, because m_locked_location.has_value()` would be equivalent.